path-manager/app/controllers/PathManagerController.scala

```diff
@@ -28,37 +28,55 @@ object PathManagerController extends Controller {
   }
 
   def registerExistingPath(id: Long) = Action { request =>
-    request.body.asJson.map(_.as[PathRecord]).map { submission =>
-      if (id != submission.identifier) {
+    val maybeSubmission = request.body.asJson.map(_.as[PathRecord])
+    maybeSubmission match {
```

Comment:
do we need this `val`?
```suggestion
    request.body.asJson.map(_.as[PathRecord]) match {
```

Reply:
Nope.